lib/puppet/pops/parser/locator.rb

```diff
@@ -262,6 +266,28 @@ def extract_text(offset, length)
   # strings are frozen).
   #
   class Locator19 < AbstractLocator
+    include Types::PuppetObject
+
+    def self._ptype
+      @type
+    end
+
+    def self.register_ptype(loader, ir)
+      locator_def = <<-OBJECT
+    {
+      attributes => {
+        string => String[1],
+        file => String,
+        line_index => {
+          type => Optional[Array[Integer]],
+          value => undef
+        }
+      }
+    }
+      OBJECT
+      @type = Pcore.add_object('Pcore::AST::Locator', locator_def, loader).resolve(Types::TypeParser.singleton, loader)
+      ir.register_implementation('Pcore::AST::Locator', self, loader)
+    end
```

Comment:
Why is AST in the {{Pcore}} name space as opposed to say {{Puppet}} ? While it is expressed in Pcore, it isn't Pcore itself is it?